include/battle/battle_context.h

```diff
@@ -0,0 +1,294 @@
+#ifndef POKEPLATINUM_BATTLE_CONTEXT_H
+#define POKEPLATINUM_BATTLE_CONTEXT_H
+
+#include "constants/battle.h"
+#include "constants/pokemon.h"
+
+#include "overlay014/struct_ov14_0221FC20_decl.h"
+#include "overlay014/struct_ov14_0221FC20_t.h"
+#include "overlay016/struct_ov16_0224B7CC_decl.h"
+#include "overlay016/struct_ov16_02252060_decl.h"
+#include "overlay016/struct_ov16_02252060_t.h"
+#include "overlay016/struct_ov16_0225C300.h"
+
+typedef struct FieldConditions {
+    u32 weatherTurns;
+    u8 futureSightTurns[MAX_BATTLERS];
+    u8 wishTurns[MAX_BATTLERS];
+    u16 futureSightMoveNum[MAX_BATTLERS];
+    int futureSightAttackerId[MAX_BATTLERS];
+    s32 futureSightDamage[MAX_BATTLERS];
+    u8 wishTarget[MAX_BATTLERS];
+} FieldConditions;
+
+typedef struct SideConditions {
+    u32 reflectBattlerId        :2,
+        reflectTurns            :3,
+        lightScreenBattlerId    :2,
+        lightScreenTurns        :3,
+        mistBattlerId           :2,
+        mistTurns               :3,
+        safeguardBattlerId      :2,
+        safeguardTurns          :3,
+        followMeActive          :1,
+        followMeBattlerId       :2,
+        knockedOffItemBattlers  :6,
+        padding00_1D            :3;
+
+    u32 spikesLayers        :2,
+        toxicSpikesLayers   :2,
+        padding04_04        :28;
```

Comment:
I would rather we make note of padding instances like these in some capacity rather than not at all. At the very least, I want a comment here noting that there are still 28 bits to constitute a full 32-bit value.